lib/cambiatus/shop/category.ex

```diff
@@ -59,7 +61,79 @@ defmodule Cambiatus.Shop.Category do
     put_assoc(changeset, :categories, categories)
   end
 
-  def positional(query \\ Category) do
+  def validate_position(changeset) do
+    position = get_field(changeset, :position)
+
+    if position < 0 do
+      add_error(changeset, :position, "position cant be negative")
+    else
+      changeset
+    end
+  end
+
+  # Do nothing if there is a parent
+  def validate_root_position(%{changes: %{parent_id: _}} = changeset), do: changeset
+
+  # When inserting
+  def validate_root_position(
+        %{changes: %{community_id: community_id, position: position}} = changeset
+      ) do
+    count = Shop.count_categories(community_id)
+
+    unless position <= count + 1 do
+      add_error(
+        changeset,
+        :position,
+        "for new categories, position must be smaller or equal than #{count + 1}"
+      )
+    else
+      changeset
+    end
```

Comment:
Talvez seja pq eu não sou acostumado com elixir, mas acho que ficaria mais legível com um `if`:

```suggestion
    if position > count + 1 do
      add_error(
        changeset,
        :position,
        "for new categories, position must be smaller or equal than #{count + 1}"
      )
    else
      changeset
    end
```

Reply:
feito!